src/core/Akka.Remote/Endpoint.cs

```diff
@@ -256,59 +263,71 @@ public OversizedPayloadException(string msg)
-    /// </remarks>
     /// </summary>
-    internal class ReliableDeliverySupervisor : UntypedActor
+    internal class ReliableDeliverySupervisor : ReceiveActor
     {
+        #region Internal message classes
+
+        public class IsIdle
+        {
+            public static readonly IsIdle Instance = new IsIdle();
+            private IsIdle() { }
+        }
+
+        public class Idle
+        {
+            public static readonly Idle Instance = new Idle();
+            private Idle() { }
+        }
+
+        public class TooLongIdle
+        {
+            public static readonly TooLongIdle Instance = new TooLongIdle();
+            private TooLongIdle() { }
+        }
+
+        #endregion
+
         private readonly ILoggingAdapter _log = Context.GetLogger();
 
-        private AkkaProtocolHandle handleOrActive;
         private readonly Address _localAddress;
         private readonly Address _remoteAddress;
         private readonly int? _refuseUid;
         private readonly AkkaProtocolTransport _transport;
         private readonly RemoteSettings _settings;
-        private AkkaPduCodec codec;
+        private AkkaPduCodec _codec;
         private AkkaProtocolHandle _currentHandle;
         private readonly ConcurrentDictionary<EndpointManager.Link, EndpointManager.ResendState> _receiveBuffers;
-        private EndpointRegistry _endpoints = new EndpointRegistry();
 
         public ReliableDeliverySupervisor(
-            AkkaProtocolHandle handleOrActive, 
-            Address localAddress, 
+            AkkaProtocolHandle handleOrActive,
+            Address localAddress,
             Address remoteAddress,
-            int? refuseUid, 
-            AkkaProtocolTransport transport, 
-            RemoteSettings settings, 
-            AkkaPduCodec codec, 
+            int? refuseUid,
+            AkkaProtocolTransport transport,
+            RemoteSettings settings,
+            AkkaPduCodec codec,
             ConcurrentDictionary<EndpointManager.Link, EndpointManager.ResendState> receiveBuffers)
         {
-            this.handleOrActive = handleOrActive;
             _localAddress = localAddress;
             _remoteAddress = remoteAddress;
             _refuseUid = refuseUid;
             _transport = transport;
             _settings = settings;
-            this.codec = codec;
+            _codec = codec;
             _currentHandle = handleOrActive;
             _receiveBuffers = receiveBuffers;
-            Reset();
-            _writer = CreateWriter();
+            Reset(); // needs to be called at startup
+            _writer = CreateWriter(); // need to create writer at startup
             Uid = handleOrActive != null ? (int?)handleOrActive.HandshakeInfo.Uid : null;
             UidConfirmed = Uid.HasValue;
+            Receiving();
+            _autoResendTimer = Context.System.Scheduler.ScheduleTellOnceCancelable(_settings.SysResendTimeout, Self, new AttemptSysMsgRedelivery(),
```

Comment:
Oh snap! this is a bug - needs to be `Context.System.Scheduler.ScheduleTellCancelableRepeatedly`